core/simulations/helpers/promptToGetNumber.ts

```diff
@@ -0,0 +1,31 @@
+import prompts from 'prompts';
+
+import hre from 'hardhat';
+
+interface BlockNumberPromt {
+    title?: string;
+    initial?: number;
+    max?: number;
+    min?: number;
+}
+const promptToGetNumber = async (params?: BlockNumberPromt) => {
+    const title: string = params?.title || 'Block number to fork from';
+    const min = params?.min || 0;
+    const initial = params?.initial;
+    const max = params?.initial;
+    // only compute latest block if we need to fill values
+    const latestBlock = max && initial ? undefined : await hre.ethers.provider.getBlockNumber();
```

Comment:
Providing `initial` shouldn't automatically set max to it as well (currently, `params.max` is unused)

```suggestion
    const max = params?.max ?? await hre.ethers.provider.getBlockNumber();
```